src/app/country/show-city/show-city.component.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+import { ShowCityComponent } from './show-city.component';
+import { Country } from '../../models/country';
+
+describe('ShowCityComponent', () => {
+  let component: ShowCityComponent;
+  let fixture: ComponentFixture<ShowCityComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ShowCityComponent],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ShowCityComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('country should be undefined', () => {
+    expect(component.country).toBeUndefined();
+    const divSelectCity = fixture.debugElement.query(By.css('.select-country'));
+    expect(divSelectCity.nativeElement.textContent.trim()).toBe('Select a country');
+  });
+
+  it('should be shown capital Unknown', () => {
+    component.country = { city: null, country: 'Spain' } as Country;
```

Comment:
Inputs should never be tested in this way. If a component has inputs they should be tested with ``TestHostComponent``. For Example:

```
@Component({
  template: `
    <dashboard-hero
      [hero]="hero">
    </dashboard-hero>`
})
class TestHostComponent {
  hero: Hero = {id: 42, name: 'Test Name'};
}
```
Then, with an instance of the test host, you can change the inputs as you want.

Reply:
true, I changed it, thanks!